src/tools/spectest-interp.cc

```diff
@@ -491,121 +613,236 @@ wabt::Result JSONParser::ParseTypeVector(TypeVector* out_types) {
 }
 
 wabt::Result JSONParser::ParseConst(TypedValue* out_value) {
-  std::string type_str;
-  std::string value_str;
-  EXPECT("{");
-  PARSE_KEY_STRING_VALUE("type", &type_str);
-  EXPECT(",");
-  PARSE_KEY_STRING_VALUE("value", &value_str);
-  EXPECT("}");
+  ExpectedValue expected;
+  CHECK_RESULT(ParseExpectedValue(&expected));
```

Comment:
Are all the consts in the JSON expected values?  What about const argument values?

Reply:
No, the expected values are more expressive. It's nice to be able to share the code since it's a significant amount of code. But yeah, I'll add an enum to make sure we don't parse stuff that isn't valid.